### ISSUE TITLE
plugins/nomad:add an option to pass a service stanza

### ISSUE THREAD

**Issue by attachmentgenie** [hashicorp/waypoint#567]
**Is your feature request related to a problem? Please describe.**
When running nomad behind a load balancer like traefik, you need to have a service stanza in place with appropriate tags for traefik to be able to route requests to your application.

**Describe the solution you'd like**
for the nomad plugin to expose some config KV pair(s) that allows the operator to specify at least 1 service (name, tags and check)

**Describe alternatives you've considered**
implementing #516 would of course allow the operator to construct what ever info is needed for allowing the app to thrive in the platform it lands on

**Comment by evanphx** [comment]
Would you mind take a moment and writing up same example hcl of what you'd like this to look like? That will help us make sure we're hitting your user case properly. Thanks!

**Comment by jeffwecan** [comment]
If it helps at all, I had sketched out a rough idea of the same sort of idea here: https://github.com/jeffwecan/waypoint/commit/c71774ab3fd39fcf3db4c6c3ece9b8775ea37189. (Though the service check bit in that example is hard-coded and it just sets the service name to the Waypoint app name rather than expose a parameter.)

Corresponding waypoint.hcl config:

```hcl
{
# ...

  deploy {
    use "nomad" {
      datacenter   = "cosmic-pup"
      service_port = 8080
      service_tags = [
        "traefik.enable=true",
        "traefik.http.routers.jorbd.tls=true",
        "traefik.http.routers.jorbd.rule=Host(`<domain>`)",
      ]
    }
  }
}

```

Part of the reason I didn't sketch out a service check parameter is I wasn't sure what sort of data type it should be represented as. Specifically whether it would make sense to abstract away some of the available options or be maximally flexible and basically duplicate all the options nomad provides.

**Comment by attachmentgenie** [comment]
job "blog" {
  datacenters = ["nomad"]
  type = "service"

  group "hugo" {
    count = 1

    task "www" {
      driver = "docker"

      config {
        image = $REGISTRY/blog:v1.0"
        port_map {
          http = 80
        }
      }

      service {
        name = "blog"
        tags = [
          $GIT_HASH_OR_SOME_OTHER_VERSION_IDENTIFIER
          "traefik.enable=true",
          "traefik.http.routers.blog.rule=Host(`blog.example.com`)"
        ]
        port = "http"
        check {
          type     = "tcp"
          interval = "10s"
          timeout  = "2s"
        }
      }
    }
  }
}

@evanphx the code above is a stripped down version of what i would like/need to deploy for for traefik to pickup the deploy using consul service discovery.

to be honest the check can probably just be a tcp check too

**Comment by attachmentgenie** [comment]
@jeffwecan thanks for taking a stab at this, your code does seem to produce the consul service i am looking for.

**Comment by rugwirobaker** [comment]
@evanphx is there any update on this😊?

**Comment by rugwirobaker** [comment]
Shouldn't these tags be specified in a release stanza instead. Where for instance you could use consul to pick the new service version and expose to the interested parties (load Balancers

**Comment by briancain** [comment]
Hey all! Thanks for all of the extra context here. This is just a heads up that we have plans to support https://github.com/hashicorp/waypoint/issues/516 and other similar issues like this one in the 0.4 milestone around a feature we're calling mutable deployments. For now I'll be updating the milestone, but I wanted to check in to let you all know we are currently working on a good solution for this and haven't forgot about this 😄 thanks!

**Comment by mitchellh** [comment]
Fixed by #1299
